pkg/authentication/bootstrapauthenticator/bootstrap.go

```diff
@@ -151,3 +151,8 @@ func (b *bootstrapUserDataGetter) Get() (*BootstrapUserData, bool, error) {
 		UID:          base64.RawURLEncoding.EncodeToString(uidBytes[:]),
 	}, true, nil
 }
+
+func (b *bootstrapUserDataGetter) IsEnabled() (bool, error) {
+	_, state, err := b.Get()
```

Comment:
I think this will need more love, maybe moving the logic that decides whether the bootstrap IdP has been removed outside of `Get()` to a separate unexported function that would be shared between these two methods.

Also the comment above mentions considering the oauth-server running inside kube-apiserver, although you were problably right to disregard that part of it, since I don't see how that would be an option these days.